docs/source/common/production_inference.rst

```diff
@@ -1,48 +1,246 @@
 
-You can also skip passing the input sample if the ` example_input_array ` property is specified in your LightningModule.
+You can also skip passing the input sample if the ``example_input_array`` property is specified in your :class:`~pytorch_lightning.core.lightning.LightningModule`.
+
+.. code-block:: python
+
+    class SimpleModel(LightningModule):
+        def __init__(self):
+            super().__init__()
+            self.l1 = torch.nn.Linear(in_features=64, out_features=4)
+            self.example_input_array = torch.randn(7, 64)
+
+        def forward(self, x):
+            return torch.relu(self.l1(x.view(x.size(0), -1)))
+
+
+    # create the model
+    model = SimpleModel()
+    filepath = "model.onnx"
+    model.to_onnx(filepath, export_params=True)
 
 Once you have the exported model, you can run it on your ONNX runtime in the following way:
 
 .. code-block:: python
 
+    import onnxruntime
+
     ort_session = onnxruntime.InferenceSession(filepath)
     input_name = ort_session.get_inputs()[0].name
-    ort_inputs = {input_name: np.random.randn(1, 64).astype(np.float32)}
+    ort_inputs = {input_name: np.random.randn(1, 64)}
     ort_outs = ort_session.run(None, ort_inputs)
 
+------------
 
-Exporting to TorchScript
-------------------------
+Convert to TorchScript
+======================
 
-TorchScript allows you to serialize your models in a way that it can be loaded in non-Python environments.
-The LightningModule has a handy method :meth:`~pytorch_lightning.core.lightning.LightningModule.to_torchscript`
-that returns a scripted module which you can save or directly use.
+`TorchScript <https://pytorch.org/docs/stable/jit.html>`_ allows you to serialize your models in a way that it can be loaded in non-Python environments.
+The ``LightningModule`` has a handy method :meth:`~pytorch_lightning.core.lightning.LightningModule.to_torchscript` that returns a scripted module which you
+can save or directly use.
 
 .. code-block:: python
 
+    class SimpleModel(LightningModule):
+        def __init__(self):
+            super().__init__()
+            self.l1 = torch.nn.Linear(in_features=64, out_features=4)
+
+        def forward(self, x):
+            return torch.relu(self.l1(x.view(x.size(0), -1)))
+
+
+    # create the model
     model = SimpleModel()
     script = model.to_torchscript()
 
     # save for use in production environment
     torch.jit.save(script, "model.pt")
 
 It is recommended that you install the latest supported version of PyTorch to use this feature without limitations.
+
+Once you have the exported model, you can run it in Pytorch or C++ runtime:
+
+.. code-block:: python
+
+    inp = torch.rand(1, 64)
+    scripted_module = torch.jit.load("model.pt")
+    output = scripted_module(dummy_input)
+
+------------
+
+PyTorch Runtime
+===============
+
+You can also load the saved checkpoint and use it as a regular :class:`torch.nn.Module`. You can extract all your :class:`torch.nn.Module`
```

Comment:
Let's talk about the best practices. They shouldn't be more than 1 direct children for the lightning module.